### PR TITLE
Clarify type declaration language in Associated Types docs

### PR DESCRIPTION
**Pull request by urschrei**
A small fix for the Associated Types docs

r? @steveklabnik

### CODE REVIEW

File: src/doc/book/associated-types.md [hunk 1]
```diff
@@ -67,7 +67,7 @@ trait Graph {
 Simple enough. Associated types use the `type` keyword, and go inside the body
 of the trait, with the functions.
 
-These `type` declarations can have all the same thing as functions do. For example,
```
Comment (tshepang) [inline src/doc/book/associated-types.md]: I would:

> These `type` declarations work the same way as those for functions.

File: src/doc/book/associated-types.md [hunk 2]
```diff
@@ -67,7 +67,7 @@ trait Graph {
...
+These `type` declarations work in the same way as those for functions do. For example,
 if we wanted our `N` type to implement `Display`, so we can print the nodes out,
```
Comment (tshepang) [inline src/doc/book/associated-types.md]: I would:

> These `type` declarations work the same way as those for functions.

Comment (steveklabnik) [inline src/doc/book/associated-types.md]: I would:

> These type declarations work in the same way as they do for functions.

Thoughts, both of you? The original sentence is very bad :(

Comment (urschrei) [inline src/doc/book/associated-types.md]: I think @tshepang's version is clearer (I assumed the original was just written late at night / in a state of exhaustion)

**Comment by steveklabnik** [comment]
@bors: r+ rollup

thank you!

**Pull request merged**